sample_annotator/click_groups_test.py

```diff
@@ -0,0 +1,29 @@
+import click
+
+
+@click.group()
+def root():
+    pass
+
+
+@click.command()
+def from_submissions():
+    click.echo("would parse submissions")
+
+
+@click.command()
+def from_tsv():
+    click.echo("would parse tsv")
+
+
+@click.command()
+def from_sqlite():
+    click.echo("would parse sqlite")
+
+
+root.add_command(from_submissions)
+root.add_command(from_tsv)
+root.add_command(from_sqlite)
+
+if __name__ == "__main__":
+    root()
```

Comment:
Can create a file called `__main__.py` instead of click_groups_test.py if the purpose of the script is to combine CLI commands from other sub modules.

A better structure to follow while making click applications is the following. Consider the below example:

```
.
├── click-package/
│   ├── __init__.py
│   ├── __main__.py
│   ├── config/
│   │   ├── __init__.py
│   │   ├── config.py # assume there's a method in this module named `show`
│   │   └── commands.py
│   ├── orchestrator/
│   └── <other-packages>
├── setup.py
└── README.md
```

`commands.py` which should be part of every submodule looks like this:

```
import click

from . import show


@click.group() # this defines my subcommand — i.e., `wfinterop config ...`
def config():
    """
    Manage app configuration.
    """
    pass


@config.command('show')
def show_config():
    """
    Show current testbed/orchestrator configuration.
    """
    show()
```

`__main__.py` should look like this:

```
#!/usr/bin/env python
import os
import sys
import logging
import pkg_resources

import click

from .config import commands as config_cli # get config commands
from .orchestrator import commands as run_cli # get orchestrator commands

logger = logging.getLogger()
logger.info("Starting `package`")


@click.group()
@click.option('--quiet', 'verbosity', flag_value='quiet',
              help=("Only display printed outputs in the console - "
                    "i.e., no log messages."))
@click.option('--debug', 'verbosity', flag_value='debug',
              help="Include all debug log messages in the console.")
def main(verbosity):
    """
    Command line interface for the `wfinterop` library.
    """
    if verbosity == 'quiet':
        logger.setLevel(logging.ERROR)
    elif verbosity == 'debug':
        logger.setLevel(logging.DEBUG)
    else:
        logger.setLevel(logging.INFO)


main.add_command(config_cli.config) # add config commands
main.add_command(run_cli.run) # add orchestrator commands


if __name__ == '__main__':
    main()
```